numpy/core/src/multiarray/methods.c

```diff
@@ -1406,16 +1406,23 @@ _deepcopy_call(char *iptr, char *optr, PyArray_Descr *dtype,
 static PyObject *
 array_deepcopy(PyArrayObject *self, PyObject *args)
 {
+    PyArrayObject* copied_array;
     PyObject* visit;
-    char *optr;
-    PyArrayIterObject *it;
+
+    NpyIter* iter;
+    NpyIter_IterNextFunc* iternext;
+
+    char* data;
+    char** dataptr;
+    npy_intp* strideptr,* innersizeptr;
+    npy_intp stride, count;
+
     PyObject *copy, *deepcopy;
-    PyArrayObject *ret;
 
     if (!PyArg_ParseTuple(args, "O", &visit)) {
         return NULL;
     }
-    ret = (PyArrayObject *)PyArray_NewCopy(self, NPY_KEEPORDER);
+    copied_array = (PyArrayObject*) PyArray_NewCopy(self, NPY_KEEPORDER);
```

Comment:
We should be checking that `copied_array` is not `NULL`, and if it is, return `NULL` to signal an error happened.